File: Pod/Classes/FNKHistogramBucketData.m
//
//  FNKBucketData.m
//  Pods
//
//  Created by Phillip Connaughton on 6/16/15.
//
//

#import "FNKHistogramBucketData.h"

@implementation FNKHistogramBucketData

-(id) init
{
    if(self == [super init])
    {
        self.data = @(0);
    }
    
    return self;
}

-(void)addDate:(NSDate*)date
{
    if(self.minDate == nil || [date compare:self.minDate] == NSOrderedAscending)
    {
        self.minDate = date;
    }
    
    if(self.maxDate == nil || [date compare:self.maxDate] == NSOrderedAscending)
    {
        self.maxDate = date;
    }
}

@end

File: Pod/Classes/FNKHistogramBucketData.swift
//
//  FNKHistogramBucketData.swift
//  Pods
//
//  Created by Phillip Connaughton on 11/21/15.
//
//

import Foundation

class FNKHistogramBucketData
{
    let data: Float
    var minDate: NSDate?
    var maxDate: NSDate?
    
    
    init(data: Float)
    {
        self.data = data
    }
    
    func addDate(date: NSDate)
    {
        if(self.minDate == nil || date.compare(self.minDate!) == .OrderedAscending)
        {
            self.minDate = date;
        }
        
        if(self.maxDate == nil || date.compare(self.maxDate!) == .OrderedAscending)
        {
            self.maxDate = date;
        }
    }
}